src/Microsoft.Identity.Client/WsTrust/WsTrustWebRequestManager.cs

```diff
@@ -139,12 +139,33 @@ public async Task<MexDocument> GetMexDocumentAsync(string federationMetadataUrl,
         {
             requestContext.Logger.Info("Sending request to userrealm endpoint.");
 
-            var uri = new UriBuilder(userRealmUriPrefix + userName + "?api-version=1.0").Uri;
+            UriBuilder builder = CreateUserrealmEndpoint(userRealmUriPrefix, userName, requestContext);
 
-            var httpResponse = await _httpManager.SendGetAsync(uri, null, requestContext).ConfigureAwait(false);
+            var httpResponse = await _httpManager.SendGetAsync(builder.Uri, null, requestContext).ConfigureAwait(false);
             return httpResponse.StatusCode == System.Net.HttpStatusCode.OK 
                 ? JsonHelper.DeserializeFromJson<UserRealmDiscoveryResponse>(httpResponse.Body) 
                 : null;
         }
-    }
+
+        public UriBuilder CreateUserrealmEndpoint(
+            string userRealmUriPrefix, 
+            string userName, 
+            RequestContext requestContext)
+        {
+            var clientInfoParameters = new Dictionary<string, string>();
+
+            foreach (KeyValuePair<string, string> kvp in MsalIdHelper.GetMsalIdParameters(requestContext.Logger))
```

Comment:
Why do you need another dictionary here? GetMsalIdParameters already returns an IDictionary<string, string>